modin/core/execution/client/io.py

```diff
@@ -0,0 +1,142 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""The module holds the factory which performs I/O using pandas on a Client."""
+
+from modin.core.io.io import BaseIO
+import fsspec
+import pandas
+
+
+class ClientIO(BaseIO):
+    """Factory providing methods for performing I/O operations using a given Client as the execution engine."""
+
+    _server_conn = None
+    _data_conn = None
+
+    @classmethod
+    def set_server_connection(cls, conn):
+        """
+        Set the server connection for the I/O object.
+
+        Parameters
+        ----------
+        conn : Any
+            Connection object that implements various methods.
+        """
+        cls._server_conn = conn
+
+    @classmethod
+    def set_data_connection(cls, conn):
+        """
+        Set the data connection for the I/O object.
+
+        Parameters
+        ----------
+        conn : Any
+            Connection object that is implementation specific.
+        """
+        cls._data_conn = conn
+
+    @classmethod
+    def read_csv(cls, filepath_or_buffer, **kwargs):
+        """
+        Read CSV data from given filepath or buffer.
+
+        Parameters
+        ----------
+        filepath_or_buffer : str, path object or file-like object
+            `filepath_or_buffer` parameter of read functions.
+        **kwargs : dict
+            Parameters of ``read_csv`` function.
+
+        Returns
+        -------
+        self.query_compiler_cls
+            Query compiler with CSV data read in.
+        """
+        if isinstance(filepath_or_buffer, str):
+            filepath_or_buffer = fsspec.open(filepath_or_buffer).full_name
+            file_protocol = "file://"
+            if filepath_or_buffer.startswith(file_protocol):
+                # We will do this so that the backend can know whether this
+                # is a path or a URL.
+                filepath_or_buffer = filepath_or_buffer[len(file_protocol) :]
```

Comment:
Is there anyway we can do this in a less janky manner? I think we can use `fsspec` to parse things out using `split_protocol`, right?